@here/harp-mapview/lib/Utils.ts

```diff
@@ -242,53 +243,182 @@ export namespace MapViewUtils {
+        camera.position.add(rotationTargetWorld);
+    }
+
+    /**
+     * @hidden
+     * @internal
+     *
+     * Clamps the supplied `deltaTilt` to the `maxTiltAngle` supplied. Note, when a non-zero offset
+     * is applied, we apply another max angle of 89 degrees to the rotation center to prevent some
+     * corner cases where the angle at the rotation center is 90 degrees and therefore intersects
+     * the geometry with the near plane.
+     */
+    function computeClampedDeltaTilt(
+        mapView: MapView,
+        offsetX: number,
+        offsetY: number,
+        deltaTilt: number,
+        maxTiltAngle: number,
+        mapTargetWorld: THREE.Vector3,
+        rotationTargetWorld: THREE.Vector3
+    ): number {
+        const camera = mapView.camera;
+        const projection = mapView.projection;
+
+        const tilt = extractTiltAngleFromLocation(projection, camera, mapTargetWorld);
+        if (tilt + deltaTilt < 0) {
+            // Clamp the final tilt to 0
+            return -tilt;
+        } else if (deltaTilt <= 0) {
+            // Reducing the tilt isn't clamped (apart from above).
+            return deltaTilt;
+        }
+
+        const rotationCenterTilt =
+            offsetX === 0 && offsetX === offsetY
+                ? tilt
+                : extractTiltAngleFromLocation(projection, camera, rotationTargetWorld!);
+
+        // This is used to find the max tilt angle, because the difference in normals is needed
+        // to correct the triangle used to find the max tilt angle at the rotation center, please
+        // read the comment above `maxTilt` below to understand why this is just an approximation.
+        let angleBetweenNormals = 0;
+        if (projection === sphereProjection) {
+            const rotationTargetNormal = projection.surfaceNormal(
+                rotationTargetWorld,
+                cache.vector3[0]
+            );
+            const mapTargetNormal = projection.surfaceNormal(mapTargetWorld, cache.vector3[1]);
+            angleBetweenNormals = rotationTargetNormal.angleTo(mapTargetNormal);
+        }
+
+        const ninetyRad = THREE.MathUtils.degToRad(90);
+        // The following terminology will be used:
+        // R = rotation target, i.e. the point about which we are rotating: `rotationTargetWorld`
+        // C = camera position
+        // M = map target, i.e. the point which the camera is looking at: `mapTargetWorld`
+        // The following variable `angleBetweenRotationTargetAndMapTarget` is the angle between the
+        // two rays C->R and C->M, i.e. the angle RCM. This angle remains constant when tilting with
+        // a fixed `offsetX` and `offsetY`. It is calculated by using the intersection of the two
+        // ways with the earth. Where:
+        // MRC is 90 + `angleBetweenNormals` - tilt at R.
+        // CMR is 90 + `tilt`
+        // RCM is 180 - CRM - RMC, aka `angleBetweenRotationTargetAndMapTarget`
```

Comment:
These formulas  don't seem to work. See a counterexample:
https://www.geogebra.org/classic/grpzmz9v

Reply:
Very cool demo, I will have to remember this tool for next time.

I see the problem, I solved the problem only for R above M, i.e. `offsetY` > 0.

I will fix, thanks a lot for spotting.

Reply:
Thinking about this a bit more, if `offsetY` is < 0, then I can just return maxTiltAngle, because then the tilt is guaranteed to be less at M.

Reply:
I added a comment to help those that may arrive at the same conclusion you did.

Reply:
> Thinking about this a bit more, if `offsetY` is < 0, then I can just return maxTiltAngle, because then the tilt is guaranteed to be less at M.

Besides the issue that this doesn't work for globe (see my new comment), I don't understand why this change fixes my review comment. Isn't the triangle calculation still wrong? I don't see why it would hold for offsetY > 0.